src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets

```diff
@@ -4,7 +4,11 @@
     <!-- Define the name of the runtime specific compiler package to import -->
     <_hostOS>$(NETCoreSdkPortableRuntimeIdentifier.SubString(0, $(NETCoreSdkPortableRuntimeIdentifier.LastIndexOf('-'))))</_hostOS>
 
+    <_targetsNonPortableSdkRid>false</_targetsNonPortableSdkRid>
+    <_targetsNonPortableSdkRid Condition="'$(RuntimeIdentifier)' == '$(NETCoreSdkRuntimeIdentifier)' and '$(RuntimeIdentifier)' != '$(NETCoreSdkPortableRuntimeIdentifier)'">true</_targetsNonPortableSdkRid>
```

Comment:
This is conflating RID of the SDK with the RID of the target (same on line 24) - so crossbuild would not work for non-portable rids. We do support crossbuild in limited scenarios: https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/cross-compile. I guess it's fine if it won't work for non-portable builds, but it should be done deliberately, so raising it.

Reply:
For non-portable rids, the only intended scenario is building for the rid itself.